worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/io/legacycompat/BannerBlockCompatibilityHandler.java

```diff
@@ -0,0 +1,150 @@
+        } catch (NullPointerException | IllegalArgumentException | ClassCastException e) {
+            tempFacing = null;
+            tempRotation = null;
+        }
+        FacingProperty = tempFacing;
+        RotationProperty = tempRotation;
+    }
+
+    @Override
+    public <B extends BlockStateHolder<B>> boolean isAffectedBlock(B block) {
+        return block.getBlockType() == BlockTypes.WHITE_BANNER
+                || block.getBlockType() == BlockTypes.WHITE_WALL_BANNER;
+    }
+
+    @Override
+    public <B extends BlockStateHolder<B>> B updateNBT(B block, Map<String, Tag> values) {
+        Tag typeTag = values.get("Base");
+        if (typeTag instanceof IntTag) {
+            boolean isWall = block.getBlockType() == BlockTypes.WHITE_WALL_BANNER;
+            String bannerType = convertBannerType(((IntTag) typeTag).getValue(), isWall);
+            if (bannerType != null) {
+                BlockType type = BlockTypes.get("minecraft:" + bannerType);
+                if (type != null) {
+                    BlockState state = type.getDefaultState();
+
+                    if (isWall) {
+                        Property facingProp = type.getProperty("facing");
+                        state = state.with(facingProp, block.getState(FacingProperty));
+                    } else {
+                        Property rotationProp = type.getProperty("rotation");
+                        state = state.with(rotationProp, block.getState(RotationProperty));
+                    }
+
+                    values.remove("Base");
+
+                    Tag patternsTag = values.get("Patterns");
+                    if (patternsTag instanceof ListTag) {
+                        List<Tag> tempList = new ArrayList<>();
+                        for (Tag pattern : ((ListTag) patternsTag).getValue()) {
+                            if (pattern instanceof CompoundTag) {
+                                Map<String, Tag> patternMap = ((CompoundTag) pattern).getValue();
+                                Tag colorTag = patternMap.get("Color");
+
+                                CompoundTagBuilder builder = CompoundTagBuilder.create();
+                                builder.putAll(patternMap);
+                                if (colorTag instanceof IntTag) {
+                                    builder.putInt("Color", 15 - ((IntTag) colorTag).getValue());
+                                }
+                                tempList.add(builder.build());
+                            } else {
+                                tempList.add(pattern);
+                            }
+                        }
+                        values.put("Patterns", new ListTag(((ListTag) patternsTag).getType(), tempList));
+                    }
+                    return (B) state;
+                }
+            }
+        }
+        return block;
+    }
+
+    private String convertBannerType(Integer oldType, boolean isWall) {
+        switch (oldType) {
```

Comment:
Can't you just get the color on the switch (or use a map/list), and append "wall_banner"/"banner" afterwards? Would be much simpler.

Reply:
Could we. Yes. 
I chose this implementation strictly for consistency with existing code. Not that I prefer this over other implementations.
https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/io/legacycompat/SkullBlockCompatibilityHandler.java#L82

Functionally it would end up doing the same thing. I would have no problem altering the PR to reflect that. Although, "Simpler" seems like an opinion in this case. The simplest is to accept the PR as presented since it works. But the Best may be to request an alteration for better code quality. Not to get caught up on word semantics.

Reply:
Sorry, I didn't leave anything there to reply to. Is this definitely a change we want? It is an easy change, but I do not desire to bother with it unless it is actually wanted.

Reply:
Skull one is like that because 2 are skulls and 4 are heads. All of these are banners. Not a huge deal either way though imo.